errai-navigation/src/main/java/org/jboss/errai/ui/nav/client/local/PageShowing.java

```diff
@@ -41,6 +43,14 @@
  * not all history token key names are known at compile time, so
  * {@code @PageState} fields can't be declared to accept their values.
  * <p>
+ * The target method is permitted an optional parameter of type {@link NavigationControl}. If the
+ * parameter is present, the page navigation will not be carried out until
+ * {@link NavigationControl#proceed()} is invoked. This is useful for interrupting page navigation
```

Comment:
The example here makes doesn't fit as well for `@PageShowing` as for `@PageHiding`. It might be better as, "This is useful for redirecting navigation before a page has been displayed, based on asynchronous logic (for example, to redirect to another page based on the result of an Errai RPC)."